pkg/controller/redirect/index.go

```diff
@@ -18,20 +18,62 @@ import (
 	"fmt"
 	"net/http"
 	"strings"
+	"time"
 
 	"github.com/google/exposure-notifications-verification-server/pkg/controller"
 )
 
+// cacheTTL is the amount of time for which to cache these values.
+const cacheTTL = 30 * time.Minute
+
 func (c *Controller) HandleIndex() http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-
 		path := strings.ToLower(r.URL.RequestURI())
 		host := strings.ToLower(r.Host)
 		// Strip of the port if that was passed along in the host header.
 		if i := strings.Index(host, ":"); i > 0 {
 			host = host[0:i]
 		}
 
+		// Check for browser type.
+		onAndroid := strings.Contains(strings.ToLower(r.UserAgent()), "android")
+		onIOS := strings.Contains(strings.ToLower(r.UserAgent()), "iphone")
+
+		// Bounce to self if we haven't already (on Android only).
+		if onAndroid && r.FormValue("bounce") == "" {
```

Comment:
Will this be a form value or a URL query param coming back?

Reply:
FormValue favors explicit forms in PUT/POST requests, but falls back to query params, which is all you got for GET.

Reply:
Huh TIL. Thanks!